fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/CoordinatorRequestBatch.java

```diff
@@ -223,19 +254,87 @@ public void addStopReplicaRequestForTabletServers(
                         });
     }
 
-    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+    /**
+     * Add updateMetadata request for tabletServers when these cases happen:
+     *
+     * <ol>
+     *   <li>case1: coordinatorServer re-start to re-initial coordinatorContext
+     *   <li>case2: Table create and bucketAssignment generated, case will happen for new created
+     *       none-partitioned table
+     *   <li>case3: Table create and bucketAssignment don't generated, case will happen for new
+     *       created partitioned table
+     *   <li>case4: Table is queued for deletion, in this case we will set a empty tableBucket set
+     *       and tableId set to {@link TableMetadata#TABLE_DURATION_DELETE_ID} to avoid send unless
+     *       info to tabletServer
+     *   <li>case5: Partition create and bucketAssignment of this partition generated.
+     *   <li>case6: Partition is queued for deletion, in this case we will set a empty tableBucket
+     *       set and partitionId set to {@link PartitionMetadata#PARTITION_DURATION_DELETE_ID } to
+     *       avoid send unless info to tabletServer
+     *   <li>case7: Leader and isr is changed for these input tableBuckets
+     *   <li>case8: One newly tabletServer added into cluster
+     *   <li>case9: One tabletServer is removed from cluster
+     * </ol>
+     */
     public void addUpdateMetadataRequestForTabletServers(
             Set<Integer> tabletServers,
-            Optional<ServerInfo> coordinatorServer,
-            Set<ServerInfo> aliveTabletServers) {
+            @Nullable Long tableId,
+            @Nullable Long partitionId,
+            Set<TableBucket> tableBuckets) {
+        // case9:
         tabletServers.stream()
                 .filter(s -> s >= 0)
-                .forEach(
-                        id ->
-                                updateMetadataRequestTabletServerSet.put(
-                                        id,
-                                        makeUpdateMetadataRequest(
-                                                coordinatorServer, aliveTabletServers)));
+                .forEach(updateMetadataRequestTabletServerSet::add);
+
+        if (tableId != null) {
+            if (partitionId != null) {
+                // case6
+                updateMetadataRequestPartitionMap
+                        .computeIfAbsent(tableId, k -> new HashMap<>())
+                        .put(partitionId, Collections.emptyList());
+            } else {
+                // case3, case4
+                updateMetadataRequestBucketMap.put(tableId, Collections.emptyList());
```

Comment:
can we explictly introduce a field like `delete_table` in our update metadata request instead of using a specifical id? I did spend some time to figure out it means.

Reply:
Ok, I will open an issue to trace this work: https://github.com/alibaba/fluss/issues/981